test/unit/util_spec.js

```diff
@@ -289,33 +289,31 @@ describe("util", function () {
   });
 
   describe("createPromiseCapability", function () {
-    it("should resolve with correct data", function (done) {
+    it("should resolve with correct data", async function () {
       const promiseCapability = createPromiseCapability();
       expect(promiseCapability.settled).toEqual(false);
 
       promiseCapability.resolve({ test: "abc" });
 
-      promiseCapability.promise.then(function (data) {
-        expect(promiseCapability.settled).toEqual(true);
-
-        expect(data).toEqual({ test: "abc" });
-        done();
-      }, done.fail);
+      const data = await promiseCapability.promise;
+      expect(promiseCapability.settled).toEqual(true);
+      expect(data).toEqual({ test: "abc" });
     });
 
-    it("should reject with correct reason", function (done) {
+    it("should reject with correct reason", async function () {
       const promiseCapability = createPromiseCapability();
       expect(promiseCapability.settled).toEqual(false);
 
       promiseCapability.reject(new Error("reason"));
 
-      promiseCapability.promise.then(done.fail, function (reason) {
+      try {
+        await promiseCapability.promise;
+        throw new Error("Promise should be rejected.");
```

Comment:
*Looking at this I realize that I've made the same mistake recently, in the API unit-tests, so we probably need to figure out a proper solution here.*

Won't this `throw new Error(...)` actually be handled by the `catch` below, which is decidedly *not* what you actually want here!?
My idea, and it's not perfect, is that we could deal with these types of situations by adding a *custom* Error-type for the unit-tests (placed in test/unit/test_utils.js) and inheriting from `BaseException`.
Maybe name it `TestShouldFailException`, better suggestions are *very* welcome, and instead do something like this (similar in other places):
```javascript
try {
  await promiseCapability.promise;
  throw new TestShouldFailException("Promise should be rejected.");
} catch (reason) {
  if (reason instanceof TestShouldFailException) {
    throw reason;
  }
  // The rest of the catch goes here...
}
```

Reply:
I'm a bit on the fence here. On one hand this is explicit, on the other hand it's extra code that we'll most likely have to repeat quite a number of times without too much added value since even if it's handled by the current `catch` the test will fail because the message doesn't agree...

Reply:
> [...] on the other hand it's extra code that we'll most likely have to repeat quite a number of times without too much added value

That's sort of the point here :-)
I see *a lot* of added value in being explicit, since it's then entirely clear what the expected behaviour is and it will *definitely* help prevent future bugs.

> [...] even if it's handled by the current `catch` the test will fail because the message doesn't agree...

While it currently "works", it's however *very* brittle and there's a very real risk that a unit-test could start failing without us noticing.
I think that requiring the `message` matching is an anti-pattern, since we cannot expect that every test will actually want to compare the messages; and in these cases we're *supposed* to fail immediately (as we're currently doing).

Reply:
That's fair indeed. Another approach I have been before is this: https://github.com/mozilla/pdf.js/blob/ebeb3f79996df5f6ff411e38e10a572157656f96/test/unit/api_spec.js#L242-L243

It's also not the best, but it would fail immediately without having to add new exception types and handling them.

Reply:
Oh, that does seem much more light-weight and it conveys the same intent (I'd forgotten about those cases); please use that pattern instead :-)